### PR TITLE
Mission control search vessel

### PR DESCRIPTION
**Pull request by louptheron**
- Search vessel from mission action panel
- Add fleet segment in from based on selected vessel

----

- [ ] Tests E2E (Cypress)

### CODE REVIEW

**Comment by louptheron** [comment]
https://github.com/MTES-MCT/monitorfish/pull/1619 must be merge before

File: frontend/src/features/vessel_sidebar/VesselSidebarHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        {isVesselNameShown && (
+          <VesselName focusOnVesselSearchInput={() => dispatch(setIsFocusedOnVesselSearch(true))} />
+        )}
+        {!isVesselNameShown && (
+          <VesselSearch
+            extendedWidth={500}
+            hasVesselIdInResults={false}
+            isExtended={isFocusedOnVesselSearch || vesselSidebarIsOpen}
+            isLastSearchedVesselsShowed={isFocusedOnVesselSearch || vesselSidebarIsOpen}
+            onClickOutsideOrEscape={() => dispatch(setIsFocusedOnVesselSearch(false))}
```
Comment (ivangabriele) [inline frontend/src/features/vessel_sidebar/VesselSidebarHeader/index.tsx:57]: I know it's cumbersome for small calls like that but unmemoized functions like that will always trigger renderings. Rerendering in general is not a big issue when you don't have performance issues with an app.

But we do have serious performance issues in our case :X. So everything that can be optimized rendering-wise, it's better to take it.

**Review by ivangabriele — APPROVED** [review]

File: frontend/src/domain/entities/vessel/riskFactor/index.ts [hunk 2]
```diff
@@ -110,3 +112,9 @@ export const getControlRateRiskFactorText = (controlRate: number) => {
 
   return undefined
 }
+
+export function getFaoZonesFromSpeciesOnboard(speciesOnboard: Array<Species>) {
+  const faoZones = speciesOnboard.map(species => species.faoZone)
+
+  return [...new Set(faoZones)]
```
[the remark this answers is not in the document]
Comment (louptheron) [reply frontend/src/domain/entities/vessel/riskFactor/index.ts:119]: it's negligible for the array length we have here (~5000) I think

File: frontend/src/features/SideWindow/MissionForm/ActionForm/SeaControl/FormikVesselSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import type { PartialSeaControl } from '../../types'
+
+export type FormikVesselSearchProps = {
+  baseRef: RefObject<HTMLDivElement>
+}
+export function FormikVesselSearch({ baseRef }: FormikVesselSearchProps) {
+  const [field, , helper] = useField<PartialSeaControl['vessel']>('vessel')
+
+  const onSelectVessel = useCallback(
+    (_nextValue: VesselIdentity) => {
```
[the remark this answers is not in the document]
Comment (louptheron) [reply frontend/src/features/SideWindow/MissionForm/ActionForm/SeaControl/FormikVesselSearch.tsx:16]: I can rewrite the nextValue variable to be less generic to remove this underscore

**Pull request merged**